### PR TITLE
Added more receivers less senders. Receivers should block.

### PR DESCRIPTION
**Pull request by kavyasrinet**

### CODE REVIEW

File: paddle/framework/channel_test.cc [hunk 1]
```diff
@@ -125,3 +125,39 @@ TEST(Channel, UnbufferedLessReceiveMoreSendTest) {
...
+    }
+  });
+  for (int i = 0; i < 5; i++) {
+    ch->Send(&i);
+    sum_send += i;
+  }
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));  // wait 0.5 sec
+  EXPECT_EQ(sum_send, 10U);
+  EXPECT_EQ(sum_receive, 10U);
+  EXPECT_EQ(sum_send, sum_receive);
```
Comment (chengduoZH) [inline paddle/framework/channel_test.cc]: I think `EXPECT_EQ(sum_send, sum_receive);` is uncessary.

File: paddle/framework/channel_test.cc [hunk 2]
```diff
@@ -125,3 +125,39 @@ TEST(Channel, UnbufferedLessReceiveMoreSendTest) {
...
+  for (int i = 5; i < 8; i++) {
+    ch->Send(&i);
+    sum_send += i;
+  }
+
+  CloseChannel(ch);
+  t.join();
+  EXPECT_EQ(sum_send, 28U);
+  EXPECT_EQ(sum_receive, 28U);
+  EXPECT_EQ(sum_send, sum_receive);
```
Comment (chengduoZH) [inline paddle/framework/channel_test.cc]: The same above.

File: paddle/framework/channel_test.cc (hunk 1 above)
Comment (kavyasrinet) [reply paddle/framework/channel_test.cc]: Thanks for pointing this out, will fix this in a commit.

File: paddle/framework/channel_test.cc (hunk 2 above)
Comment (kavyasrinet) [reply paddle/framework/channel_test.cc]: Fixing this as well. Thanks!

**Pull request merged**